buildconfig/Jenkins/conda-setup

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+# This script expects a environment variable called $WORKSPACE to be set.
+
+# From https://stackoverflow.com/a/9107028 by https://stackoverflow.com/users/1184238/andrew-norrie
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+
+EXPECTED_MAMBAFORGE_PATH=$HOME/mambaforge
+EXPECTED_CONDA_PATH=$EXPECTED_MAMBAFORGE_PATH/bin/conda
+ENV_NAME="mantid-developer"
+
+MAMBAFORGE_SCRIPT_NAME=Mambaforge-$(uname)-$(uname -m).sh
+
+# Ensure conda is installed
+if [[ ! -f $EXPECTED_CONDA_PATH ]]; then
+    if [[ ! -f $MAMBAFORGE_SCRIPT_NAME ]]; then
+        # Download mambaforge
+        URL=https://github.com/conda-forge/miniforge/releases/latest/download/$MAMBAFORGE_SCRIPT_NAME
+        if [ -x "$(which curl)" ]; then
+            curl -L -O $URL
+        elif [ -x "$(which wget)" ] ; then
+            wget $URL
+        else
+            echo "Could not download Conda as wget and curl are not installed."
+        fi
+    fi
+    bash $MAMBAFORGE_SCRIPT_NAME -b
+    rm $MAMBAFORGE_SCRIPT_NAME
+fi
+
+# With miniconda we can grab the mantid environment if it doesn't already exist
```

Comment:
I think this script is doing a little too much. We've mixed up the download and install miniconda/mambaforge with creating an environment. Can we split things up to scripts:
* `download-and-install-conda`: Accepts an argument for where to install it and an optional flag to say whether it should overwrite an existing installation.
* `create-conda-env`: Accepts the name of the conda environment with a flag to say whether it should forcibly overwrite an existing env